sdks/python/apache_beam/runners/portability/fn_api_runner/translations.py

```diff
@@ -347,6 +348,22 @@ def add_or_get_coder_id(self,
     self.components.coders[new_coder_id].CopyFrom(coder_proto)
     return new_coder_id
 
+  def add_data_channel_coder(self, pcoll_id):
+    pcoll = self.components.pcollections[pcoll_id]
+    proto = beam_runner_api_pb2.Coder(
+        spec=beam_runner_api_pb2.FunctionSpec(
+            urn=common_urns.coders.WINDOWED_VALUE.urn),
+        component_coder_ids=[
+            pcoll.coder_id,
+            self.components.windowing_strategies[
+                pcoll.windowing_strategy_id].window_coder_id
+        ])
+    channel_coder = self.add_or_get_coder_id(
+        proto, pcoll.coder_id + '_windowed')
+    if pcoll.coder_id in self.safe_coders:
+      channel_coder = self.length_prefixed_coder(channel_coder)
```

Comment:
Note this check is already done within `length_prefix_coder` where it will return the original coder id if its a safe coder so you 

```suggestion
    channel_coder = self.length_prefixed_coder(channel_coder)
```

Reply:
Here I want to populate the safe_coders mapping for channel_coder iff the value coder has such a mapping.

Reply:
lines 398-399 already do this check as part of the `length_prefixed_coder` method.

Reply:
I now understand the nuance of your statement.

Reply:
Yes, but here `pcoll.coder_id != channel_coder`.